app/src/crash/crash-app.tsx

```diff
@@ -0,0 +1,176 @@
+  readonly error?: Error
+  readonly windowState: WindowState
+}
+
+// Note that we're reusing the welcome illustrations here, any changes to these
+// will have to be reflected in the welcome flow as well.
+const BottomImageUri = `file:///${__dirname}/static/welcome-illustration-left-bottom.svg`
+
+function prepareErrorMessage(error: Error) {
+
+  let message
+
+  if (error.stack) {
+    message = error.stack
+      .split('\n')
+      .map((line) => {
+        // The stack trace lines come in two forms:
+        //
+        // `at Function.module.exports.Emitter.simpleDispatch (SOME_USER_SPECIFIC_PATH/app/node_modules/event-kit/lib/emitter.js:25:14)`
+        // `at file:///SOME_USER_SPECIFIC_PATH/app/renderer.js:6:4250`
+        //
+        // We want to try to strip the user-specific path part out.
+        const match = line.match(/(\s*)(.*)(\(|file:\/\/\/).*(app.*)/)
+
+        return !match || match.length < 5
+          ? line
+          : match[1] + match[2] + match[3] + match[4]
+      })
+      .join('\n')
+  } else {
+    message = `${error.name}: ${error.message}`
+  }
+
+  return `${message}\n\nVersion: ${getVersion()}\nOS: ${getOS()}\n`
+
+}
+
+export class CrashApp extends React.Component<ICrashAppProps, ICrashAppState> {
+
+  public constructor(props: ICrashAppProps) {
+    super(props)
+
+    this.state = {
+      windowState: getWindowState(remote.getCurrentWindow()),
+    }
+  }
+
+  public componentDidMount() {
+    const window = remote.getCurrentWindow()
+
+    ipcRenderer.on(windowStateChannelName, (_, args) => {
+      this.setState({ windowState: getWindowState(window) })
+    })
+
+    ipcRenderer.on('error', (event: Electron.IpcRendererEvent, crashDetails: ICrashDetails) => {
+      this.setState(crashDetails)
+    })
+
+    const now = Date.now()
+    ipcRenderer.send('crash-ready', now - this.props.startTime)
+  }
+
+  private onIssueTrackerLinkClicked = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    shell.openExternal('https://github.com/desktop/desktop/issues')
+    e.preventDefault()
+  }
+
+  private onQuitButtonClicked = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault()
+    ipcRenderer.send('crash-quit')
+  }
+
+  private renderTitle() {
+    const message = this.state.type === 'launch'
+      ? 'GitHub Desktop failed to launch'
+      : 'GitHub Desktop encountered an error'
+
+    return (
+      <header>
+        <Octicon symbol={OcticonSymbol.stop} className='error-icon' />
+        <h1>{message}</h1>
+      </header>
+    )
+  }
+
+  private renderDescription() {
+    if (this.state.type === 'launch') {
+      return (
+        <p>
+          GitHub Desktop encountered a catastrophic error that prevents it from
+          launching. This has been reported to the team, but if you encounter this
+          repeatedly please report this issue to the
+          GitHub Desktop <a href='https://github.com/desktop/desktop/issues' onClick={this.onIssueTrackerLinkClicked}>issue tracker</a>.
```

Comment:
I think I was just trying to pull in as little as possible from the renderer in case the originating error resided in any of the components we pull in but `LinkButton` seems safe enough